### ISSUE TITLE
Ordering of installation of vscode:extensions matters and leads to errors on loading

### ISSUE THREAD

**Issue by krukow** [microsoft/vscode#196794]
Behaviour:
- Upon opening a GitHub code space with the following configuration of extensions, I get an error saying that one of the extensions are not installed (but it should be installed since it is in the list).

- VS Code Version: Version: 1.83.1 (Universal) / Commit: f1b07bd25dfad64b0167beb15359ae573aecd2cc
- OS Version: macOS 13.6 (22G120)

## Steps to Reproduce:

1. Use GitHub code spaces with the following .devcontainer (see below)
2. Open the GitHub code space
3. When loaded you get the error: "Cannot activate the 'autopep8' extension because it depends on the 'Python' extension, which is not loaded. Would you like to reload the window to load the extension?"

## Diagnosis:

In speaking with the GitHub code spaces team, I learned that: 

- "It looks like all of the extensions are being passed into the VS Code server at the same time, so it would require upstream changes in VS Code to handle the installation order in a better way."

and

- "My guess is that the extensions are installed in parallel and the python one just takes longer to install/activate and the pep8 one is trying to activate too early before the python finishes installing"

`devcontainer.json` includes
```json
{
	"name": "Python 3",
	"image": "mcr.microsoft.com/devcontainers/python:0-3.11",
	"features": {
	    "ghcr.io/devcontainers-contrib/features/poetry:2": {},
	    "ghcr.io/devcontainers/features/github-cli:1": {}
	},

	"customizations": {

		"vscode": {
			"extensions": [
				"ms-python.python",
				"ms-toolsai.jupyter",
				"eamodio.gitlens",

				"redhat.vscode-yaml",
				"GitHub.copilot",
				"ms-python.vscode-pylance",
				"ms-python.autopep8"
			],
			"settings": {
			    "python.testing.pytestArgs": [
			        "data"
			    ],
			    "python.testing.unittestEnabled": true,
			    "python.testing.pytestEnabled": true,
			    "[python]": {
			        "editor.defaultFormatter": "ms-python.autopep8",
			        "editor.formatOnSave": true
			    }
			}
		}
	}
}
```

**Comment by lgarron** [comment]
I've also been seeing such issues for months, particularly for extensions that depend on `ms-python.python`.

As someone maintaining the development setup environment for a team, it would be really helpful to have any solution that works without a manual reload or special configuration for each user.

**Comment by seesharprun** [comment]
We have a similar issue with C#. If you install the C# extension or the C# DevKit extension, it will install quicker than it's dependent ".NET Install Tool" extension.

This race condition leads to a prompt that shows up the majority of the time to reload the editor to install the dependent extension.

**Comment by mjpizz** [comment]
I have been running into this same situation, except inside a devcontainer. The Black Formatter extension and the Flake8 extension fail 100% of the time and VSCode suggests reloading. Reloading always works.

Is anyone aware of a workaround in the meantime to enforce an ordering?

**Comment by mjpizz** [comment]
I stumbled upon a workaround with a long-running `updateContentCommand` in .devcontainer.json, so I suspect for anyone having this issue you could potentially do something like:

```
"updateContentCommand": "sleep 20",
```

Not pretty, but neither is the "always click the reload button" option.

**Comment by Codym48** [comment]
I'm seeing the same behavior as @mjpizz with a slight clarification:
> The Black Formatter extension_, the isort extension, _and the Flake8 extension fail 100% of the time _when calling **Rebuild and Reopen in Container**,_ and VSCode suggests reloading. Reloading always works. _Just doing **Reopen in Container** also seems to always work._

I have the following in my .devcontainer.json:
```json

			"extensions": [
				"bierner.markdown-mermaid",
				"donjayamanne.githistory",
				"h5web.vscode-h5web",
				"janisdd.vscode-edit-csv",
				"jebbs.plantuml",
				"ms-azuretools.vscode-docker",
				"ms-python.black-formatter",
				"ms-python.flake8",
				"ms-python.isort",
				"ms-python.python",
				"ms-toolsai.jupyter",
				"ryanluker.vscode-coverage-gutters"
			]
```
Order (moving python above black-formatter, flake8, and isort) doesn't seem to matter.

**Comment by seesharprun** [comment]
@craiglpeters

**Comment by seesharprun** [comment]
This bug makes it rough to create a "minimal" template for .NET and Python developers with the .NET/Python extensions as it will almost always ask you to reopen the codespace.

Even this configuration will require a "reopen" prompt:

```json
{
  "customizations": {
    "vscode": {
      "extensions": [
        "ms-dotnettools.csdevkit"
      ]
    }
  }
}
```

**Comment by bilderbuchi** [comment]
> Something like an "extension-order" setting, similar to the features, would be amazing in VS Code or development containers.

Or at least loading extensions in the _order as given in the `extensions` list_, then users could at least easily hand-sort to avoid these annoying dependency problems until a "real" fix comes along. Presumably/hopefully this could be a quick fix as it's a small change?

**Comment by seesharprun** [comment]
@sandy081

### Repro steps

1. I use this button to open the [`seesharprun/codespace-test`](https://github.com/seesharprun/codespace-test) repository as a codespace:

    [![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/seesharprun/codespace-test?quickstart=1&devcontainer_path=.devcontainer%2Fdevcontainer.json)
    
    Relevant devcontainer configuration:
    
    ```json
    {
      "customizations": {
        "vscode": {
          "extensions": [
            "ms-dotnettools.csdevkit"
          ]
        }
      }
    }
    ```
    
1. I waited for the C# extensions to install and to observe the *"Cannot activate the 'C#' extension because it depends on the '.NET Install Tool' extension, which is not loaded. Would you like to reload the window to load the extension?"* popup.

1. F1 > Open View > Server

1. Here's the logs:

    ```
    2023-11-29 15:41:44.047 [trace] Remote configuration data at /home/codespace/.vscode-remote
    2023-11-29 15:41:44.047 [trace] process arguments: {"_":[],"host":"127.0.0.1","port":"0","connection-token-file":"/home/codespace/.vscode-remote/data/Machine/.connection-token-1a5daa3a0231a0fbba4f14db7ec463cf99d7768e","without-connection-token":false,"disable-websocket-compression":false,"print-startup-performance":false,"print-ip-address":false,"accept-server-license-terms":true,"server-data-dir":"/home/codespace/.vscode-remote","enable-smoke-test-driver":false,"disable-telemetry":false,"disable-workspace-trust":false,"log":["trace"],"force-disable-user-env":true,"enable-sync":false,"use-test-resolver":false,"extensions-download-dir":"/home/codespace/.vscode-remote/extensionsCache","install-extension":["ms-dotnettools.csdevkit"],"install-builtin-extension":["GitHub.vscode-pull-request-github","github.github-vscode-theme","github.codespaces"],"list-extensions":false,"show-versions":false,"force":false,"do-not-sync":true,"pre-release":false,"start-server":true,"enable-remote-auto-shutdown":true,"remote-auto-shutdown-without-delay":false,"use-host-proxy":false,"without-browser-env-var":false,"help":false,"version":false,"user-data-dir":"/home/codespace/.vscode-remote/data","builtin-extensions-dir":"/vscode/bin/linux-x64/1a5daa3a0231a0fbba4f14db7ec463cf99d7768e/extensions","extensions-dir":"/home/codespace/.vscode-remote/extensions","logsPath":"/home/codespace/.vscode-remote/data/logs/20231129T154143"}
    2023-11-29 15:41:44.047 [info] 
    
    
    
    
    2023-11-29 15:41:44.047 [trace] [File Watcher (node.js)] Request to start watching: /home/codespace/.vscode-remote/data/Machine (excludes: <none>, includes: <all>, correlationId: <none>),/home/codespace/.vscode-remote/data/Machine/settings.json (excludes: <none>, includes: <all>, correlationId: <none>)
    2023-11-29 15:41:44.047 [trace] Installing builtin extensions passed via args...
    2023-11-29 15:41:44.047 [trace] Started scanning user extensions {"$mid":1,"external":"file:///home/codespace/.vscode-remote/extensions/extensions.json","path":"/home/codespace/.vscode-remote/extensions/extensions.json","scheme":"file"}
    2023-11-29 15:41:44.047 [trace] Installing extensions passed via args...
    2023-11-29 15:41:44.047 [info] Extension host agent started.
    2023-11-29 15:41:44.052 [trace] [File Watcher (node.js)] Started watching: '/home/codespace/.vscode-remote/data/Machine'
    2023-11-29 15:41:44.053 [trace] [File Watcher (node.js)] Request to start watching: /home/codespace/.vscode-remote/extensions (excludes: <none>, includes: <all>, correlationId: <none>)
    2023-11-29 15:41:44.059 [trace] [File Watcher (node.js)] Started watching: '/home/codespace/.vscode-remote/data/Machine/settings.json'
    2023-11-29 15:41:44.061 [info] Started initializing default profile extensions in extensions installation folder. file:///home/codespace/.vscode-remote/extensions
    2023-11-29 15:41:44.061 [trace] Started scanning user extensions {"$mid":1,"fsPath":"/home/codespace/.vscode-remote/extensions","external":"file:///home/codespace/.vscode-remote/extensions","path":"/home/codespace/.vscode-remote/extensions","scheme":"file"}
    2023-11-29 15:41:44.076 [trace] Deleting from cache /home/codespace/.vscode-remote/extensionsCache/formulahendry.code-runner-0.12.1
    2023-11-29 15:41:44.077 [trace] Deleting from cache /home/codespace/.vscode-remote/extensionsCache/vscjava.vscode-java-debug-0.55.0
    2023-11-29 15:41:44.077 [trace] Deleting from cache /home/codespace/.vscode-remote/extensionsCache/ms-toolsai.jupyter-2023.11.1003351000
    2023-11-29 15:41:44.078 [trace] Deleting from cache /home/codespace/.vscode-remote/extensionsCache/vscode-icons-team.vscode-icons-12.6.0
    2023-11-29 15:41:44.081 [trace] [File Watcher (node.js)] Started watching: '/home/codespace/.vscode-remote/extensions'
    2023-11-29 15:41:44.086 [debug] ComputeTargetPlatform: linux-x64
    2023-11-29 15:41:44.086 [trace] Scanned user extensions: 0
    2023-11-29 15:41:44.088 [trace] [File Watcher (node.js)] [raw] ["rename"] extensions.json
    2023-11-29 15:41:44.088 [trace] [File Watcher (node.js)] [raw] ["change"] extensions.json
    2023-11-29 15:41:44.089 [trace] [File Watcher (node.js)] [ADDED] /home/codespace/.vscode-remote/extensions/extensions.json
    2023-11-29 15:41:44.136 [info] Completed initializing default profile extensions in extensions installation folder. file:///home/codespace/.vscode-remote/extensions
    2023-11-29 15:41:44.138 [trace] [File Watcher (node.js)] Request to start watching: /home/codespace/.vscode-remote/extensions/extensions.json (excludes: <none>, includes: <all>, correlationId: <none>)
    2023-11-29 15:41:44.142 [trace] Scanned user extensions: 0
    2023-11-29 15:41:44.142 [trace] [File Watcher (node.js)] Started watching: '/home/codespace/.vscode-remote/extensions/extensions.json'
    2023-11-29 15:41:44.144 [info] [127.0.0.1][2bf0f2be][ManagementConnection] New connection established.
    2023-11-29 15:41:44.145 [trace] Started scanning user extensions {"$mid":1,"fsPath":"/home/codespace/.vscode-remote/extensions","external":"file:///home/codespace/.vscode-remote/extensions","path":"/home/codespace/.vscode-remote/extensions","scheme":"file"}
    2023-11-29 15:41:44.153 [debug] ComputeTargetPlatform: linux-x64
    2023-11-29 15:41:44.155 [trace] resolveShellEnv(): skipped (--force-disable-user-env)
    2023-11-29 15:41:44.222 [trace] [File Watcher (node.js)]  >> normalized [ADDED] /home/codespace/.vscode-remote/extensions/extensions.json
    2023-11-29 15:41:44.227 [trace] [File Watcher (node.js)] [CHANGED] /home/codespace/.vscode-remote/extensions/extensions.json
    2023-11-29 15:41:44.232 [trace] Scanned user extensions: 0
    2023-11-29 15:41:44.281 [trace] Started scanning system extensions
    2023-11-29 15:41:44.282 [trace] Started scanning user extensions {"$mid":1,"fsPath":"/home/codespace/.vscode-remote/extensions/extensions.json","external":"file:///home/codespace/.vscode-remote/extensions/extensions.json","path":"/home/codespace/.vscode-remote/extensions/extensions.json","scheme":"file"}
    2023-11-29 15:41:44.282 [trace] Started scanning system extensions
    2023-11-29 15:41:44.283 [trace] Started scanning user extensions {"$mid":1,"fsPath":"/home/codespace/.vscode-remote/extensions/extensions.json","external":"file:///home/codespace/.vscode-remote/extensions/extensions.json","path":"/home/codespace/.vscode-remote/extensions/extensions.json","scheme":"file"}
    2023-11-29 15:41:44.283 [trace] ExtensionManagementService.refreshReportedCache
    2023-11-29 15:41:44.284 [trace] resolveShellEnv(): skipped (--force-disable-user-env)
    2023-11-29 15:41:44.284 [trace] Started scanning system extensions
    2023-11-29 15:41:44.291 [trace] Started scanning user extensions {"$mid":1,"fsPath":"/home/codespace/.vscode-remote/extensions/extensions.json","external":"file:///home/codespace/.vscode-remote/extensions/extensions.json","path":"/home/codespace/.vscode-remote/extensions/extensions.json","scheme":"file"}
    2023-11-29 15:41:44.294 [info] Log level changed to info
    2023-11-29 15:41:45.641 [info] [127.0.0.1][db55662e][ExtensionHostConnection] New connection established.
    2023-11-29 15:41:45.675 [info] [127.0.0.1][db55662e][ExtensionHostConnection] <1742> Launched Extension Host Process.
    2023-11-29 15:41:46.267 [info] Installing builtin extension 'github.github-vscode-theme'...
    2023-11-29 15:41:46.276 [info] Getting Manifest... github.github-vscode-theme
    2023-11-29 15:41:46.300 [info] Installing builtin extension 'github.vscode-pull-request-github'...
    2023-11-29 15:41:46.321 [info] Getting Manifest... github.vscode-pull-request-github
    2023-11-29 15:41:46.335 [info] Installing builtin extension 'github.codespaces'...
    2023-11-29 15:41:46.336 [info] Getting Manifest... github.codespaces
    2023-11-29 15:41:46.431 [info] Installing extension: github.github-vscode-theme
    2023-11-29 15:41:46.435 [info] Installing extension: github.codespaces
    2023-11-29 15:41:46.448 [info] Installing extension: github.vscode-pull-request-github
    2023-11-29 15:41:47.529 [warning] Extension signature verification failed with error 'ENOENT': github.github-vscode-theme
    2023-11-29 15:41:47.547 [warning] Extension signature verification failed with error 'ENOENT': github.codespaces
    2023-11-29 15:41:47.677 [info] Extracted extension to file:///home/codespace/.vscode-remote/extensions/github.github-vscode-theme-6.3.4: github.github-vscode-theme
    2023-11-29 15:41:47.700 [info] Renamed to /home/codespace/.vscode-remote/extensions/github.github-vscode-theme-6.3.4
    2023-11-29 15:41:47.706 [info] Extracting extension completed. github.github-vscode-theme
    2023-11-29 15:41:47.726 [warning] Extension signature verification failed with error 'ENOENT': github.vscode-pull-request-github
    2023-11-29 15:41:47.734 [info] Extension installed successfully: github.github-vscode-theme
    2023-11-29 15:41:47.735 [info] Extension 'github.github-vscode-theme' v6.3.4 was successfully installed.
    2023-11-29 15:41:47.956 [info] Extracted extension to file:///home/codespace/.vscode-remote/extensions/github.vscode-pull-request-github-0.76.1: github.vscode-pull-request-github
    2023-11-29 15:41:47.982 [info] Renamed to /home/codespace/.vscode-remote/extensions/github.vscode-pull-request-github-0.76.1
    2023-11-29 15:41:47.988 [info] Extracting extension completed. github.vscode-pull-request-github
    2023-11-29 15:41:48.013 [info] Extension installed successfully: github.vscode-pull-request-github
    2023-11-29 15:41:48.016 [info] Extension 'github.vscode-pull-request-github' v0.76.1 was successfully installed.
    2023-11-29 15:41:48.172 [info] Extracted extension to file:///home/codespace/.vscode-remote/extensions/github.codespaces-1.16.2: github.codespaces
    2023-11-29 15:41:48.192 [info] Renamed to /home/codespace/.vscode-remote/extensions/github.codespaces-1.16.2
    2023-11-29 15:41:48.196 [info] Extracting extension completed. github.codespaces
    2023-11-29 15:41:48.213 [info] Extension installed successfully: github.codespaces
    2023-11-29 15:41:48.215 [info] Extension 'github.codespaces' v1.16.2 was successfully installed.
    2023-11-29 15:41:48.216 [info] Installing extensions...
    2023-11-29 15:41:48.645 [info] Installing extension 'ms-dotnettools.csdevkit'...
    2023-11-29 15:41:48.646 [info] Getting Manifest... ms-dotnettools.csdevkit
    2023-11-29 15:41:48.665 [info] Installing extension: ms-dotnettools.csdevkit
    2023-11-29 15:41:49.241 [info] Getting Manifest... ms-dotnettools.csharp
    2023-11-29 15:41:49.360 [info] Getting Manifest... ms-dotnettools.vscode-dotnet-runtime
    2023-11-29 15:41:49.472 [info] Getting Manifest... ms-dotnettools.vscodeintellicode-csharp
    2023-11-29 15:41:49.566 [info] Getting Manifest... ms-dotnettools.csharp
    2023-11-29 15:41:49.706 [info] Installing extension: ms-dotnettools.csharp
    2023-11-29 15:41:49.706 [info] Installing the extension without checking dependencies and pack ms-dotnettools.csharp
    2023-11-29 15:41:49.752 [info] Installing extension: ms-dotnettools.csharp ms-dotnettools.csdevkit
    2023-11-29 15:41:49.754 [info] Installing extension: ms-dotnettools.vscode-dotnet-runtime ms-dotnettools.csdevkit
    2023-11-29 15:41:49.754 [info] Installing extension: ms-dotnettools.vscodeintellicode-csharp ms-dotnettools.csdevkit
    2023-11-29 15:41:50.556 [warning] Extension signature verification failed with error 'ENOENT': ms-dotnettools.vscode-dotnet-runtime
    2023-11-29 15:41:50.712 [warning] Extension signature verification failed with error 'ENOENT': ms-dotnettools.csharp
    2023-11-29 15:41:50.973 [info] Extracted extension to file:///home/codespace/.vscode-remote/extensions/ms-dotnettools.vscode-dotnet-runtime-2.0.0: ms-dotnettools.vscode-dotnet-runtime
    2023-11-29 15:41:51.002 [info] Renamed to /home/codespace/.vscode-remote/extensions/ms-dotnettools.vscode-dotnet-runtime-2.0.0
    2023-11-29 15:41:51.010 [info] Extracting extension completed. ms-dotnettools.vscode-dotnet-runtime
    2023-11-29 15:41:53.781 [info] Extracted extension to file:///home/codespace/.vscode-remote/extensions/ms-dotnettools.csharp-2.0.251: ms-dotnettools.csharp
    2023-11-29 15:41:53.820 [info] Renamed to /home/codespace/.vscode-remote/extensions/ms-dotnettools.csharp-2.0.251
    2023-11-29 15:41:53.830 [info] Extracting extension completed. ms-dotnettools.csharp
    2023-11-29 15:41:53.848 [info] Extension installed successfully: ms-dotnettools.csharp
    2023-11-29 15:41:54.874 [info] Getting Manifest... github.vscode-pull-request-github
    2023-11-29 15:41:55.008 [info] Installing extension: github.vscode-pull-request-github
    2023-11-29 15:41:55.008 [info] Installing the extension without checking dependencies and pack github.vscode-pull-request-github
    2023-11-29 15:41:55.247 [warning] Extension signature verification failed with error 'ENOENT': ms-dotnettools.csharp
    2023-11-29 15:41:55.313 [info] Extension installed successfully: github.vscode-pull-request-github
    2023-11-29 15:41:56.006 [info] Getting Manifest... ms-dotnettools.vscode-dotnet-runtime
    2023-11-29 15:41:56.028 [info] Extensions is already requested to install ms-dotnettools.vscode-dotnet-runtime
    2023-11-29 15:41:56.739 [info] Getting Manifest... github.codespaces
    2023-11-29 15:41:56.762 [info] Installing extension: github.codespaces
    2023-11-29 15:41:56.762 [info] Installing the extension without checking dependencies and pack github.codespaces
    2023-11-29 15:41:57.272 [info] Extension installed successfully: github.codespaces
    2023-11-29 15:41:58.010 [info] Getting Manifest... github.vscode-github-actions
    2023-11-29 15:41:58.038 [info] Installing extension: github.vscode-github-actions
    2023-11-29 15:41:58.040 [info] Installing the extension without checking dependencies and pack github.vscode-github-actions
    2023-11-29 15:41:58.973 [warning] Extension signature verification failed with error 'ENOENT': github.vscode-github-actions
    2023-11-29 15:42:00.258 [info] Extracted extension to file:///home/codespace/.vscode-remote/extensions/github.vscode-github-actions-0.26.2: github.vscode-github-actions
    2023-11-29 15:42:00.472 [info] Renamed to /home/codespace/.vscode-remote/extensions/github.vscode-github-actions-0.26.2
    2023-11-29 15:42:00.554 [info] Extracting extension completed. github.vscode-github-actions
    2023-11-29 15:42:00.729 [info] Extension installed successfully: github.vscode-github-actions
    2023-11-29 15:42:04.255 [info] Getting Manifest... github.vscode-github-actions
    2023-11-29 15:42:04.460 [info] Getting Manifest... ms-dotnettools.vscodeintellicode-csharp
    2023-11-29 15:42:04.545 [info] Getting Manifest... ms-dotnettools.csdevkit
    2023-11-29 15:42:04.608 [info] Getting Manifest... ms-dotnettools.csharp
    2023-11-29 15:42:04.643 [info] Installing extension: github.vscode-github-actions
    2023-11-29 15:42:04.643 [info] Installing the extension without checking dependencies and pack github.vscode-github-actions
    2023-11-29 15:42:04.645 [info] Extensions is already requested to install ms-dotnettools.vscodeintellicode-csharp
    2023-11-29 15:42:04.645 [info] Extensions is already requested to install ms-dotnettools.csdevkit
    2023-11-29 15:42:04.646 [info] Installing extension: ms-dotnettools.csharp
    2023-11-29 15:42:04.646 [info] Installing the extension without checking dependencies and pack ms-dotnettools.csharp
    2023-11-29 15:42:05.508 [info] Extension installed successfully: github.vscode-github-actions
    2023-11-29 15:42:08.049 [warning] Extension signature verification failed with error 'ENOENT': ms-dotnettools.csharp
    2023-11-29 15:42:22.900 [info] Extracted extension to file:///home/codespace/.vscode-remote/extensions/ms-dotnettools.csharp-2.10.28-linux-x64: ms-dotnettools.csharp
    2023-11-29 15:42:22.941 [info] Renamed to /home/codespace/.vscode-remote/extensions/ms-dotnettools.csharp-2.10.28-linux-x64
    2023-11-29 15:42:22.953 [info] Extracting extension completed. ms-dotnettools.csharp
    2023-11-29 15:42:23.454 [info] Marked extension as uninstalled ms-dotnettools.csharp-2.0.251
    2023-11-29 15:42:24.391 [warning] Extension signature verification failed with error 'ENOENT': ms-dotnettools.csdevkit
    2023-11-29 15:42:28.945 [info] Extracted extension to file:///home/codespace/.vscode-remote/extensions/ms-dotnettools.csharp-2.12.4-linux-x64: ms-dotnettools.csharp
    2023-11-29 15:42:29.172 [info] Extracted extension to file:///home/codespace/.vscode-remote/extensions/ms-dotnettools.csdevkit-1.0.14-linux-x64: ms-dotnettools.csdevkit
    2023-11-29 15:42:29.254 [info] Renamed to /home/codespace/.vscode-remote/extensions/ms-dotnettools.csharp-2.12.4-linux-x64
    2023-11-29 15:42:29.266 [info] Extracting extension completed. ms-dotnettools.csharp
    2023-11-29 15:42:29.278 [info] Renamed to /home/codespace/.vscode-remote/extensions/ms-dotnettools.csdevkit-1.0.14-linux-x64
    2023-11-29 15:42:29.282 [info] Extracting extension completed. ms-dotnettools.csdevkit
    2023-11-29 15:42:29.296 [info] Extension installed successfully: ms-dotnettools.csharp
    2023-11-29 15:42:29.306 [info] Marked extension as uninstalled ms-dotnettools.csharp-2.10.28-linux-x64
    2023-11-29 15:42:30.856 [warning] Extension signature verification failed with error 'ENOENT': ms-dotnettools.vscodeintellicode-csharp
    2023-11-29 15:42:36.079 [info] Extracted extension to file:///home/codespace/.vscode-remote/extensions/ms-dotnettools.vscodeintellicode-csharp-0.1.26-linux-x64: ms-dotnettools.vscodeintellicode-csharp
    2023-11-29 15:42:36.102 [info] Renamed to /home/codespace/.vscode-remote/extensions/ms-dotnettools.vscodeintellicode-csharp-0.1.26-linux-x64
    2023-11-29 15:42:36.103 [info] Extracting extension completed. ms-dotnettools.vscodeintellicode-csharp
    2023-11-29 15:42:36.132 [info] Extension installed successfully: ms-dotnettools.vscode-dotnet-runtime
    2023-11-29 15:42:36.132 [info] Extension installed successfully: ms-dotnettools.csharp
    2023-11-29 15:42:36.132 [info] Extension installed successfully: ms-dotnettools.csdevkit
    2023-11-29 15:42:36.132 [info] Extension installed successfully: ms-dotnettools.vscodeintellicode-csharp
    2023-11-29 15:42:36.136 [info] Extension 'ms-dotnettools.csdevkit' v1.0.14 was successfully installed.
    2023-11-29 15:42:36.393 [info] Deleted uninstalled extension from disk ms-dotnettools.csharp /home/codespace/.vscode-remote/extensions/ms-dotnettools.csharp-2.10.28-linux-x64
    2023-11-29 15:42:36.440 [info] Deleted uninstalled extension from disk ms-dotnettools.csharp /home/codespace/.vscode-remote/extensions/ms-dotnettools.csharp-2.0.251
    ```

**Comment by seesharprun** [comment]
I repeated this experiment using a Codespace on a different repository and configuration file.

- Repository: [`seesharprun/codespace-test-python`](https://github.com/seesharprun/codespace-test-python)
- Codespaces link: <https://codespaces.new/seesharprun/codespace-test-python?quickstart=1&devcontainer_path=.devcontainer%2Fdevcontainer.json>
- Devcontainer configuration:

  ```json
  {
    "customizations": {
      "vscode": {
        "extensions": [
      	  "bierner.markdown-mermaid",
      	  "donjayamanne.githistory",
      	  "h5web.vscode-h5web",
      	  "janisdd.vscode-edit-csv",
      	  "jebbs.plantuml",
      	  "ms-azuretools.vscode-docker",
      	  "ms-python.black-formatter",
      	  "ms-python.flake8",
      	  "ms-python.isort",
      	  "ms-python.python",
      	  "ms-toolsai.jupyter",
      	  "ryanluker.vscode-coverage-gutters"
        ]
      }
    }
  }
  ```
  
I could not repro the same issue that @Codym48 observed. I may be missing an important step to repro.

**Comment by sandy081** [comment]
@seesharprun Thanks for the log. From the log it seems following extensions are requested to be installed 

- github.codespaces
- github.vscode-github-actions
- github.vscode-pull-request-github

and following extensions are requested to be installed again

- ms-dotnettools.vscodeintellicode-csharp
- ms-dotnettools.csdevkit
- ms-dotnettools.csharp
- ms-dotnettools.vscode-dotnet-runtime

By any chance, you have settings sync enabled and you have above extensions synced?

**Comment by seesharprun** [comment]
I do have settings sync enabled with the C# and C# DevKit extensions.

**Comment by Codym48** [comment]
Thank you for trying @seesharprun! I tried your [seesharprun/codespace-test-python](https://github.com/seesharprun/codespace-test-python) and also didn't see the issue. Given @mjpizz's [workaround](https://github.com/microsoft/vscode/issues/196794#issuecomment-1823826407), maybe the codespace takes just long enough to load things before opening the window to avoid the Reload prompt?

I reproduced the issue in a smaller repository via the Dev Containers and WSL extensions and the following container spec
```json
{
    "name": "Python 3",
    "image": "mcr.microsoft.com/devcontainers/python:0-3.11",
    "customizations": {
        "vscode": {
            "extensions": [
                "ms-python.python",
                "ms-python.black-formatter",
                "ms-python.isort",
                "ms-python.flake8"
            ]
        }
    },
    "postCreateCommand": "pip3 install black isort flake8"
}
``` 

![image](https://github.com/microsoft/vscode/assets/34431661/6207c97a-a0ca-4f0b-ad5e-3c2bc0db54e6)

Help > About:

Version: 1.68.1 (user setup)
Commit: 30d9c6cd9483b2cc586687151bcbcd635f373630
Date: 2022-06-14T12:48:58.283Z
Electron: 17.4.7
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Windows_NT x64 10.0.19045

Note this is an older version of VS Code than I used above (https://github.com/microsoft/vscode/issues/196794#issuecomment-1830356267).

F1 > Open View > Log (Remote Server)
```
[2023-12-01 04:53:14.820] [remoteagent] [info] Extension host agent started.
[2023-12-01 04:53:14.911] [remoteagent] [info] [127.0.0.1][609518d8][ManagementConnection] New connection established.
[2023-12-01 04:53:15.835] [remoteagent] [info] [127.0.0.1][ce546b6e][ExtensionHostConnection] New connection established.
[2023-12-01 04:53:15.873] [remoteagent] [info] [127.0.0.1][ce546b6e][ExtensionHostConnection] <270> Launched Extension Host Process.
[2023-12-01 04:53:17.035] [remoteagent] [info] Getting Manifest... ms-python.black-formatter
[2023-12-01 04:53:17.114] [remoteagent] [info] Getting Manifest... ms-python.isort
[2023-12-01 04:53:17.121] [remoteagent] [info] Installing extension: ms-python.black-formatter
[2023-12-01 04:53:17.295] [remoteagent] [info] Installing extension: ms-python.isort
[2023-12-01 04:53:17.408] [remoteagent] [error] Failed to install extension. ms-python.flake8
[2023-12-01 04:53:17.422] [remoteagent] [error] Incompatible: Can't install 'ms-python.flake8' extension because it is not compatible with the current version of Visual Studio Code (version 1.68.1).
    at _.checkAndGetCompatibleVersion (/vscode/vscode-server/bin/linux-x64/30d9c6cd9483b2cc586687151bcbcd635f373630/out/vs/server/node/server.main.js:102:11814)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async _.installFromGallery (/vscode/vscode-server/bin/linux-x64/30d9c6cd9483b2cc586687151bcbcd635f373630/out/vs/server/node/server.main.js:102:3200)
    at async S.installFromGallery (/vscode/vscode-server/bin/linux-x64/30d9c6cd9483b2cc586687151bcbcd635f373630/out/vs/server/node/server.main.js:81:1328)
    at async /vscode/vscode-server/bin/linux-x64/30d9c6cd9483b2cc586687151bcbcd635f373630/out/vs/server/node/server.main.js:80:141569
    at async Promise.all (index 3)
    at async S.installExtensions (/vscode/vscode-server/bin/linux-x64/30d9c6cd9483b2cc586687151bcbcd635f373630/out/vs/server/node/server.main.js:80:141484)
[2023-12-01 04:53:17.661] [remoteagent] [info] Getting Manifest... ms-python.python
[2023-12-01 04:53:17.797] [remoteagent] [info] Installing extension: ms-python.python
[2023-12-01 04:53:18.480] [remoteagent] [info] Getting Manifest... ms-python.python
[2023-12-01 04:53:19.026] [remoteagent] [info] Getting Manifest... ms-python.python
[2023-12-01 04:53:20.296] [remoteagent] [info] Getting Manifest... ms-python.vscode-pylance
[2023-12-01 04:53:21.348] [remoteagent] [info] Getting Manifest... ms-python.vscode-pylance
[2023-12-01 04:53:21.671] [remoteagent] [info] Getting Manifest... ms-toolsai.jupyter
[2023-12-01 04:53:22.165] [remoteagent] [info] Getting Manifest... ms-toolsai.jupyter-keymap
[2023-12-01 04:53:22.253] [remoteagent] [info] Getting Manifest... ms-python.vscode-pylance
[2023-12-01 04:53:22.406] [remoteagent] [info] Getting Manifest... ms-toolsai.jupyter
[2023-12-01 04:53:22.600] [remoteagent] [info] Getting Manifest... ms-toolsai.jupyter-renderers
[2023-12-01 04:53:22.984] [remoteagent] [info] Installing extension: ms-python.vscode-pylance
[2023-12-01 04:53:22.985] [remoteagent] [info] Installing extension: ms-toolsai.jupyter
[2023-12-01 04:53:22.986] [remoteagent] [info] Installing extension: ms-toolsai.jupyter-keymap
[2023-12-01 04:53:22.987] [remoteagent] [info] Installing extension: ms-toolsai.jupyter-renderers
[2023-12-01 04:53:23.349] [remoteagent] [info] Getting Manifest... ms-toolsai.jupyter-keymap
[2023-12-01 04:53:23.506] [remoteagent] [info] Downloaded extension: ms-toolsai.jupyter /root/.vscode-server/extensionsCache/ms-toolsai.jupyter-2022.5.1001601848
[2023-12-01 04:53:23.510] [remoteagent] [info] Downloaded extension: ms-toolsai.jupyter-keymap /root/.vscode-server/extensionsCache/ms-toolsai.jupyter-keymap-1.1.0
[2023-12-01 04:53:23.511] [remoteagent] [info] Downloaded extension: ms-python.python /root/.vscode-server/extensionsCache/ms-python.python-2022.16.1
[2023-12-01 04:53:23.513] [remoteagent] [info] Downloaded extension: ms-toolsai.jupyter-renderers /root/.vscode-server/extensionsCache/ms-toolsai.jupyter-renderers-1.0.9
[2023-12-01 04:53:23.626] [remoteagent] [info] Extracted extension to /root/.vscode-server/extensions/.389438c2-7eff-49b0-8a6d-de8ee8cdc05d: ms-toolsai.jupyter-keymap
[2023-12-01 04:53:23.647] [remoteagent] [info] Renamed to /root/.vscode-server/extensions/ms-toolsai.jupyter-keymap-1.1.0
[2023-12-01 04:53:23.659] [remoteagent] [info] Extracting completed. ms-toolsai.jupyter-keymap
[2023-12-01 04:53:23.729] [remoteagent] [info] Getting Manifest... ms-toolsai.jupyter-renderers
[2023-12-01 04:53:24.028] [remoteagent] [info] Extension is already requested to install ms-python.python
[2023-12-01 04:53:24.028] [remoteagent] [info] Extension is already requested to install ms-python.vscode-pylance
[2023-12-01 04:53:24.028] [remoteagent] [info] Extension is already requested to install ms-toolsai.jupyter
[2023-12-01 04:53:24.028] [remoteagent] [info] Extension is already requested to install ms-toolsai.jupyter-renderers
[2023-12-01 04:53:24.134] [remoteagent] [info] Downloaded extension: ms-python.black-formatter /root/.vscode-server/extensionsCache/ms-python.black-formatter-2022.4.0
[2023-12-01 04:53:24.396] [remoteagent] [info] Getting Manifest... ms-toolsai.jupyter
[2023-12-01 04:53:24.795] [remoteagent] [info] Getting Manifest... ms-toolsai.jupyter-keymap
[2023-12-01 04:53:25.007] [remoteagent] [info] Getting Manifest... ms-toolsai.jupyter-renderers
[2023-12-01 04:53:25.196] [remoteagent] [info] Extension is already requested to install ms-python.python
[2023-12-01 04:53:25.197] [remoteagent] [info] Extension is already requested to install ms-python.vscode-pylance
[2023-12-01 04:53:25.197] [remoteagent] [info] Extension is already requested to install ms-toolsai.jupyter
[2023-12-01 04:53:25.197] [remoteagent] [info] Extension is already requested to install ms-toolsai.jupyter-renderers
[2023-12-01 04:53:25.367] [remoteagent] [info] Downloaded extension: ms-python.isort /root/.vscode-server/extensionsCache/ms-python.isort-2022.2.0
[2023-12-01 04:53:25.933] [remoteagent] [info] Extracted extension to /root/.vscode-server/extensions/.65621849-e307-4794-99cd-de016b5bfd01: ms-toolsai.jupyter-renderers
[2023-12-01 04:53:25.954] [remoteagent] [info] Renamed to /root/.vscode-server/extensions/ms-toolsai.jupyter-renderers-1.0.9
[2023-12-01 04:53:25.967] [remoteagent] [info] Extracting completed. ms-toolsai.jupyter-renderers
[2023-12-01 04:53:27.175] [remoteagent] [info] Extracted extension to /root/.vscode-server/extensions/.09dcb16d-b955-49b2-b4db-4cd35da5f4e4: ms-python.isort
[2023-12-01 04:53:27.196] [remoteagent] [info] Renamed to /root/.vscode-server/extensions/ms-python.isort-2022.2.0
[2023-12-01 04:53:27.205] [remoteagent] [info] Extracting completed. ms-python.isort
[2023-12-01 04:53:27.206] [remoteagent] [info] Extension installed successfully: ms-python.isort
[2023-12-01 04:53:28.082] [remoteagent] [info] Extracted extension to /root/.vscode-server/extensions/.d8569ab8-99d8-4cba-aa0a-30308e1850fc: ms-python.black-formatter
[2023-12-01 04:53:28.335] [remoteagent] [info] Renamed to /root/.vscode-server/extensions/ms-python.black-formatter-2022.4.0
[2023-12-01 04:53:28.344] [remoteagent] [info] Extracting completed. ms-python.black-formatter
[2023-12-01 04:53:28.347] [remoteagent] [info] Extension installed successfully: ms-python.black-formatter
[2023-12-01 04:53:28.713] [remoteagent] [info] Extracted extension to /root/.vscode-server/extensions/.dcbee190-bcbf-4c0e-9e1b-d9db930aeba9: ms-toolsai.jupyter
[2023-12-01 04:53:28.790] [remoteagent] [info] Renamed to /root/.vscode-server/extensions/ms-toolsai.jupyter-2022.5.1001601848
[2023-12-01 04:53:28.948] [remoteagent] [info] Extracting completed. ms-toolsai.jupyter
[2023-12-01 04:53:32.370] [remoteagent] [info] Extracted extension to /root/.vscode-server/extensions/.ea3ded73-2958-4593-aedd-38be47ed159a: ms-python.python
[2023-12-01 04:53:32.416] [remoteagent] [info] Renamed to /root/.vscode-server/extensions/ms-python.python-2022.16.1
[2023-12-01 04:53:32.429] [remoteagent] [info] Extracting completed. ms-python.python
[2023-12-01 04:53:32.500] [remoteagent] [info] Downloaded extension: ms-python.vscode-pylance /root/.vscode-server/extensionsCache/ms-python.vscode-pylance-2023.1.10
[2023-12-01 04:53:39.782] [remoteagent] [info] Extracted extension to /root/.vscode-server/extensions/.01d6f240-55a4-4cd3-8dab-fb4aa3e404fc: ms-python.vscode-pylance
[2023-12-01 04:53:39.809] [remoteagent] [info] Renamed to /root/.vscode-server/extensions/ms-python.vscode-pylance-2023.1.10
[2023-12-01 04:53:39.814] [remoteagent] [info] Extracting completed. ms-python.vscode-pylance
[2023-12-01 04:53:39.816] [remoteagent] [info] Extension installed successfully: ms-toolsai.jupyter-keymap
[2023-12-01 04:53:39.817] [remoteagent] [info] Extension installed successfully: ms-toolsai.jupyter-renderers
[2023-12-01 04:53:39.817] [remoteagent] [info] Extension installed successfully: ms-toolsai.jupyter
[2023-12-01 04:53:39.817] [remoteagent] [info] Extension installed successfully: ms-python.python
[2023-12-01 04:53:39.817] [remoteagent] [info] Extension installed successfully: ms-python.vscode-pylance
[2023-12-01 04:53:39.829] [remoteagent] [error] Error: Failed Installing Extensions: ms-python.flake8
    at S.installExtensions (/vscode/vscode-server/bin/linux-x64/30d9c6cd9483b2cc586687151bcbcd635f373630/out/vs/server/node/server.main.js:81:42)
```
This is without Settings Sync enabled

![image](https://github.com/microsoft/vscode/assets/34431661/2556927e-1be3-4b46-bdf6-cc91ce3a46b2)

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!

**Comment by bilderbuchi** [comment]
> We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future.

I have to admit that I'm confused. 
The linked page talks about closing out-of-scope _feature requests_. I don't see how this can be relevant here, as this report is of extension loading being _broken/erroring_. In fact, I'm confused why the `bug` label was removed by @sandy081. 
Furthermore, users have supplied logs as requested, and even repos for reproduction, yet the `info-needed` label was not removed, and now failing extension loading is closed as `out-of-scope`?

@isidorn I'd be curious about the decision process here, to be honest. 😞 ☮️

**Comment by krukow** [comment]
Yeah - I also disagree with closing as `not planned` here. I'm continuing to see wierd bugs with extensions here - these are effective usability problems that cause friction and requires workarounds.

Example with latest version of VSCode:

```
Version: 1.85.1 (Universal)
Commit: 0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2
Date: 2023-12-13T09:48:06.308Z
Electron: 25.9.7
ElectronBuildId: 25551756
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 22.6.0
```


### Behaviour

- Upon opening a GitHub code space with the following configuration of extensions, I get an error saying that one of the extensions are not installed (but it should be installed since it is in the list).


## Steps to Reproduce:

1. Use GitHub code spaces with the following .devcontainer (see below)
2. Open the GitHub code space
3. When loaded you get the error: "Cannot activate the 'autopep8' extension because it depends on the 'Python' extension, which is not loaded. Would you like to reload the window to load the extension?"

```
// For format details, see https://aka.ms/devcontainer.json. For config options, see the
// README at: https://github.com/devcontainers/templates/tree/main/src/python
{
	"name": "Python 3",
	// Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
	"image": "mcr.microsoft.com/devcontainers/base:ubuntu",
	// Features to add to the dev container. More info: https://containers.dev/features.
	"features": {
		"ghcr.io/devcontainers/features/python:1": {
			"version": "3.12",
			"installTools": true,
			"installJupyterlab": true,
			"configureJupyterlabAllowOrigin": "*"
		},
		"ghcr.io/devcontainers-contrib/features/poetry:2": {},
		"ghcr.io/devcontainers/features/github-cli:1": {}
	},
	// Configure tool-specific properties.
	"customizations": {
		// Configure properties specific to VS Code.
		"vscode": {
			"extensions": [
				"ms-toolsai.jupyter",
				"GitHub.copilot",
				"ms-python.python",
				"ms-python.vscode-pylance",
				"ms-python.black-formatter"
			],
			"settings": {
				"[python]": {
					"editor.defaultFormatter": "ms-python.black-formatter"
				},
				"python.formatting.provider": "none"
			}
		},
		"codespaces": {
			"repositories": {
				"github/thehub": {
					"permissions": {
						"contents": "read"
					  }
				}
			}
		}
	},

	// Use 'forwardPorts' to make a list of ports inside the container available locally.
	// "forwardPorts": [9000],
	"forwardPorts": [
		8888,
		8889
	],
	
	// Use 'portsAttributes' to set default properties for specific forwarded ports. 
	// More info: https://containers.dev/implementors/json_reference/#port-attributes
	// "portsAttributes": {
	// 	"9000": {
	// 		"label": "Hello Remote World",
	// 		"onAutoForward": "notify"
	// 	}
	// },

	// Use 'postCreateCommand' to run commands after the container is created.
	// using "features" above instead
	"postCreateCommand": "git config --global --add safe.directory /workspaces/cliff && git pull --update && poetry install --no-root",
	"postStartCommand": "bash .devcontainer/postStartCommand.sh"
	// Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
	// "remoteUser": "root"
}
```

Despite clearly declaring Python extension, I get this error in the log:

```
Visual Studio Code (1.85.1, codespaces, desktop)
Jupyter Extension Version: 2023.11.1003402403.
Python Extension not installed.
Pylance Extension not installed.
```

Now if I close and reopen the codespace, I get thing working:

```
Visual Studio Code (1.85.1, codespaces, desktop)
Jupyter Extension Version: 2023.11.1003402403.
Python Extension Version: 2023.22.1.
Pylance Extension Version: 2023.12.1.
Platform: linux (x64).
Workspace folder /workspaces/cliff, Home = /home/vscode
14:05:35.243 [info] Start refreshing Kernel Picker (1702562735242)
14:05:35.262 [info] Using Pylance
14:05:36.466 [warn] Failed to get activated env vars for /usr/local/python/current/bin/python in 1040ms
14:05:36.786 [info] Process Execution: /usr/local/python/current/bin/python -c "import site;print("USER_BASE_VALUE");print(site.USER_BASE);print("USER_BASE_VALUE");"
14:05:36.838 [error] Unable to determine site packages path for python /usr/local/python/current/bin/python (Unknown)
14:05:36.873 [info] Process Execution: /usr/local/python/current/bin/python -m pip list
14:05:37.928 [info] End refreshing Kernel Picker (1702562735242)
14:05:38.417 [info] Start refreshing Kernel Picker (1702562738417)
14:05:39.557 [info] End refreshing Kernel Picker (1702562738417)
```

and my poetry managed environment is available to Jupyter.

It's really hard for my users to understand why they have to restart VSCode and reopen the codespace before the extension works.

**Comment by krukow** [comment]
I work at GitHub and I would be happy to jump on a call and walk you through how to reproduce this if you can't reproduce.

**Comment by isidorn** [comment]
Thanks for feedback. I closed this because I do not see us investing resources in fixing issues like this in the next 6 months. Not saying it is invalid. We might still reopen it after I discuss with @sandy081 

As for GitHub - I suggest to reach out to the Codespaces team and if they have capacity they can invest in making a PR.

**Comment by krukow** [comment]
any updates @isidorn or @sandy081 ?

I have another instance of this again: with `"ms-python.python",` clearly in the devcontainer

![Image](https://github.com/microsoft/vscode/assets/3635/99ee26a4-91bb-4fcd-bdf9-3a57fd59dcb4)

with a .devcontainer.json of:

```
"customizations": {
		// Configure properties specific to VS Code.
		"vscode": {
			"extensions": [
				"ms-toolsai.jupyter",
				"GitHub.copilot",
				"ms-python.python",
				"ms-python.vscode-pylance",
				"ms-python.black-formatter"
			],
```

**Comment by sandy081** [comment]
> any updates @isidorn or @sandy081 ?

Will investigate and see if I can repro

**Comment by krukow** [comment]
@sandy081 for [my example](https://github.com/microsoft/vscode/issues/196794#issuecomment-1888578602), please see server log here: https://gist.github.com/krukow/9b0b4b8f6aeca97cce7fe1e48c73112e

With `devcontainer.json` and latest version

```
// For format details, see https://aka.ms/devcontainer.json. For config options, see the
// README at: https://github.com/devcontainers/templates/tree/main/src/python
{
	"name": "Python 3",
	// Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
	"image": "mcr.microsoft.com/devcontainers/base:ubuntu",
	// Features to add to the dev container. More info: https://containers.dev/features.
	"features": {
		"ghcr.io/devcontainers/features/python:1": {
			"version": "3.12",
			"installTools": true,
			"installJupyterlab": true,
			"configureJupyterlabAllowOrigin": "*"
		},
		"ghcr.io/devcontainers-contrib/features/poetry:2": {},
		"ghcr.io/devcontainers/features/github-cli:1": {}
	},
	// Configure tool-specific properties.
	"customizations": {
		// Configure properties specific to VS Code.
		"vscode": {
			"extensions": [
				"ms-toolsai.jupyter",
				"GitHub.copilot",
				"ms-python.python",
				"ms-python.vscode-pylance",
				"ms-python.black-formatter"
			],
			"settings": {
				"[python]": {
					"editor.defaultFormatter": "ms-python.black-formatter"
				},
				"python.formatting.provider": "none"
			}
		},
		"codespaces": {
			"repositories": {
				"github/thehub": {
					"permissions": {
						"contents": "read"
					  }
				}
			}
		}
	},

	// Use 'forwardPorts' to make a list of ports inside the container available locally.
	// "forwardPorts": [9000],
	"forwardPorts": [
		8888,
		8889
	]
}
```

System info
```
Version: 1.85.1 (Universal)
Commit: 0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2
Date: 2023-12-13T09:48:06.308Z
Electron: 25.9.7
ElectronBuildId: 25551756
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 23.2.0
```

**Comment by luabud** [comment]
this [issue](https://github.com/microsoft/vscode-remote-release/issues/9426#issue-2096903066) shows another simple repro example, they even provided a repo that reproduces this issue every time and it only has two extensions listed in devcontainer.json: https://github.com/PelicanQ/devcontainer-issue.git

**Comment by PelicanQ** [comment]
As a follow-up to luabuds post above, this is the logs from F1 > Open View > Server.
```
21:46:47.776 [info] Installing extensions...
21:46:47.776 [info] Extension host agent started.
21:46:47.793 [info] Started initializing default profile extensions in extensions installation folder. file:///root/.vscode-server/extensions
21:46:47.801 [debug] ComputeTargetPlatform: linux-x64
21:46:47.809 [info] Completed initializing default profile extensions in extensions installation folder. file:///root/.vscode-server/extensions
21:46:47.815 [debug] ComputeTargetPlatform: linux-x64
21:46:47.921 [info] [127.0.0.1][f0ad4f32][ManagementConnection] New connection established.
21:46:47.940 [info] Log level changed to info
21:46:48.495 [info] [127.0.0.1][fc4073b9][ExtensionHostConnection] New connection established.
21:46:48.598 [info] [127.0.0.1][fc4073b9][ExtensionHostConnection] <275> Launched Extension Host Process.
21:46:49.404 [info] Installing extension 'ms-python.black-formatter'...
21:46:49.406 [info] Getting Manifest... ms-python.black-formatter
21:46:49.415 [info] Installing extension 'ms-python.python'...
21:46:49.415 [info] Getting Manifest... ms-python.python
21:46:49.538 [info] Installing extension: ms-python.black-formatter
21:46:49.586 [info] Installing extension: ms-python.python
21:46:50.185 [info] Getting Manifest... ms-python.python
21:46:50.346 [info] Getting Manifest... ms-python.vscode-pylance
21:46:50.562 [info] Installing extension: ms-python.vscode-pylance ms-python.python
21:46:50.788 [warning] Extension signature verification failed with error 'ENOENT': ms-python.python
21:46:50.982 [info] Getting Manifest... ms-python.vscode-pylance
21:46:51.150 [info] Waiting for already requested installing extension ms-python.python ms-python.black-formatter
21:46:51.151 [info] Waiting for already requested installing extension ms-python.vscode-pylance ms-python.black-formatter
21:46:51.309 [warning] Extension signature verification failed with error 'ENOENT': ms-python.black-formatter
21:46:52.187 [info] Extracted extension to file:///root/.vscode-server/extensions/ms-python.black-formatter-2023.6.0: ms-python.black-formatter
21:46:52.200 [info] Renamed to /root/.vscode-server/extensions/ms-python.black-formatter-2023.6.0
21:46:52.205 [info] Extracting extension completed. ms-python.black-formatter
21:46:55.111 [info] Extracted extension to file:///root/.vscode-server/extensions/ms-python.python-2023.22.1: ms-python.python
21:46:55.124 [info] Renamed to /root/.vscode-server/extensions/ms-python.python-2023.22.1
21:46:55.129 [info] Extracting extension completed. ms-python.python
21:46:55.309 [warning] Extension signature verification failed with error 'ENOENT': ms-python.vscode-pylance
21:47:03.113 [info] Extracted extension to file:///root/.vscode-server/extensions/ms-python.vscode-pylance-2023.12.1: ms-python.vscode-pylance
21:47:03.129 [info] Renamed to /root/.vscode-server/extensions/ms-python.vscode-pylance-2023.12.1
21:47:03.134 [info] Extracting extension completed. ms-python.vscode-pylance
21:47:03.143 [info] Extension installed successfully: ms-python.python
21:47:03.144 [info] Extension installed successfully: ms-python.vscode-pylance
21:47:03.148 [info] Finished waiting for already requested installing extension ms-python.python ms-python.black-formatter
21:47:03.148 [info] Finished waiting for already requested installing extension ms-python.vscode-pylance ms-python.black-formatter
21:47:03.149 [info] Extension 'ms-python.python' v2023.22.1 was successfully installed.
21:47:03.149 [info] Extension installed successfully: ms-python.black-formatter
21:47:03.150 [info] Extension 'ms-python.black-formatter' v2023.6.0 was successfully installed.
```

**Comment by krukow** [comment]
thanks @sandy081 🎉
